packages/server-core/src/user/identity-provider/identity-provider.class.ts

```diff
@@ -155,16 +158,30 @@ export class IdentityProvider extends Service {
     )
     // DRC
     try {
-      if (result.user.userRole !== 'guest') {
+      if (result.user.userRole === 'user' || result.user.userRole === 'admin') {
+        var response = await axios.post(`${BLOCKCHAIN_URL}/authorizeServer`, {
+          authSecretKey: BLOCKCHAIN_URL_SECRET
+        })
+        var accessToken = response.data.accessToken
+        var walletData = await axios.post(
```

Comment:
lets move these axios requests into appropriate abstraction functions such that we can reuse and test these easily

Reply:
changes done